src/wordlist.c

```diff
@@ -83,6 +83,24 @@ size_t wordlist_lookup_word(const struct words *w, const char *word)
     return found ? found - w->indices + 1u : 0u;
 }
 
+size_t wordlist_lookup_prefix(const struct words *w, const char *prefix)
```

Comment:
this breaks with `act` versus `actual` mnemonic words, since `act` prefix matches for 2+ entries.

I think we should restrict the `prefix` input to length 3 or 4, and if length 3 it must be an exact match.